### PR TITLE
[#768] improvement(CI): Avoiding related issues by changing CI pipeline dependencies

### PR DESCRIPTION
**Pull request by qqqttt123**
### What changes were proposed in this pull request?
Setting the job dependencies.  There are some issues if we use two workflows. I change them to one workflow.

### Why are the changes needed?

Fix: #768 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Tested in my repo.
https://github.com/qqqttt123/gravitino/pull/4

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -109,3 +109,51 @@ jobs:
...
+        with:
+          java-version: ${{ matrix.java-version }}
+          distribution: 'temurin'
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Package Gravitino
+        run: |
+          ./gradlew build -x test -PjdkVersion=${{ matrix.java-version }}
```
Comment (jerryshao) [inline .github/workflows/test.yml:138]: Is this step still required? As we've already build the project above.

Comment (yuqi1129) [reply .github/workflows/test.yml:138]: If the job name is 'Package Gravitino' and it is indeed a packaging task, the `build` task can be skipped and use `compileDistribution` directly.

**Comment by yuqi1129** [comment]
I recommend changing the title to "Avoiding Related Issues by Changing CI Pipeline Dependencies" or something similar.

**Comment by qqqttt123** [comment]
> I recommend changing the title to "Avoiding Related Issues by Changing CI Pipeline Dependencies" or something similar.

done.

File: .github/workflows/test.yml (hunk 1 above)
Comment (qqqttt123) [reply .github/workflows/test.yml]: fixed.

Comment (qqqttt123) [reply .github/workflows/test.yml]: I tried. But It will fail.

Comment (yuqi1129) [reply .github/workflows/test.yml:138]: I execute `./gradlew compileDistribution -x test` locally with executing `./gradlew build`, and it seems no error exists

Comment (qqqttt123) [reply .github/workflows/test.yml:138]: It doesn't work. You can see https://github.com/datastrato/gravitino/actions/runs/7536000721

Comment (yuqi1129) [reply .github/workflows/test.yml:138]: I wander why `./gradlew compileDistribution -x test` will trigger integration test...

Comment (qqqttt123) [reply .github/workflows/test.yml:138]: > I wander why `./gradlew compileDistribution -x test` will trigger integration test...

We have next steps. The step which you mentioned doesn't trigger.

Comment (jerryshao) [reply .github/workflows/test.yml:138]: We have already built the Gravitino in line 85, I was thinking why can't we reuse the built package?

Comment (qqqttt123) [reply .github/workflows/test.yml:138]: They're different jobs.  Different jobs may be scheduled to different machines.

**Comment by jerryshao** [comment]
One thing I was thinking is what is the benefit of this change? I can see the total CI time will be increased because we make it in sequence, with the cost of it what was the actual benefit?

**Comment by qqqttt123** [comment]
> One thing I was thinking is what is the benefit of this change? I can see the total CI time will be increased because we make it in sequence, with the cost of it what was the actual benefit?

We can save the resource of Github action. That's a good point. But for CI time, this is not a good choice.

**Pull request closed without merging**